olclient/entity_helpers/work.py

```diff
@@ -138,6 +139,25 @@ def rm_subjects(self, subjects, comment=''):
             data['subjects'] = list(set(data['subjects']) - set(subjects))
             return self.OL.session.put(url, json.dumps(data))
 
+        def _delete_editions(self, edition_olids: List[str], comment: str) -> None:
+            editions_cleanup_response: Response = self.OL.delete_many(edition_olids, comment)
+            if editions_cleanup_response.ok is False:
+                raise Exception(f'Could not delete all editions of work {self.olid}. '
+                                f'response: {editions_cleanup_response}')
+
+        def delete(self, comment: str, confirm: bool = True):
+            edition_olids: List[str] = [edition_data.olid for edition_data in self.editions]
+            edition_count = len(edition_olids)
+            should_delete = confirm is False or get_approval_from_cli(
+                f'Delete https://openlibrary.org/works/{self.olid} and its {edition_count} editions?'
+            )
+            if should_delete is False:
+                return
+
+            if edition_count > 0:
+                self._delete_editions(edition_olids, comment)
+            return self.OL.delete(self.olid)
```

Comment:
Actually, we might want to combine these together; I can't imagine it would fail in such a way where only some would be deleted. Then we also get this being a single bulk delete request
```suggestion
            return self.OL.delete_many([self.olid, *edition_olids], comment)
```

Reply:
yeah, I was concerned about a situation where if one edition fails to be deleted, then we would have a hanging edition since the work will be gone. But what you are saying makes sense too 👍🏽